site/src/pages/WorkspacePage/AppStatuses.tsx

```diff
@@ -0,0 +1,406 @@
+import type { Theme } from "@emotion/react";
+import { useTheme } from "@emotion/react";
+import AppsIcon from "@mui/icons-material/Apps";
+import CheckCircle from "@mui/icons-material/CheckCircle";
+import ErrorIcon from "@mui/icons-material/Error";
+import HelpOutline from "@mui/icons-material/HelpOutline";
+import InsertDriveFile from "@mui/icons-material/InsertDriveFile";
+import OpenInNew from "@mui/icons-material/OpenInNew";
+import Warning from "@mui/icons-material/Warning";
+import CircularProgress from "@mui/material/CircularProgress";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import type {
+	WorkspaceAppStatus as APIWorkspaceAppStatus,
+	Workspace,
+	WorkspaceAgent,
+	WorkspaceApp,
+} from "api/typesGenerated";
+import { useProxy } from "contexts/ProxyContext";
+import { formatDistance, formatDistanceToNow } from "date-fns";
+import { DividerWithText } from "pages/DeploymentSettingsPage/LicensesSettingsPage/DividerWithText";
+import type { FC } from "react";
+import { createAppLinkHref } from "utils/apps";
+
+const getStatusColor = (
+	theme: Theme,
+	state: APIWorkspaceAppStatus["state"],
+) => {
+	switch (state) {
+		case "complete":
+			return theme.palette.success.main;
+		case "failure":
+			return theme.palette.error.main;
+		case "working":
+			return theme.palette.primary.main;
+		default:
+			// Assuming unknown state maps to warning/secondary visually
+			return theme.palette.text.secondary;
+	}
+};
+
+const getStatusIcon = (
+	theme: Theme,
+	state: APIWorkspaceAppStatus["state"],
+	isLatest: boolean,
+) => {
+	// Determine color: Use state color if latest, otherwise use disabled text color (grey)
+	const color = isLatest
+		? getStatusColor(theme, state)
+		: theme.palette.text.disabled;
+	switch (state) {
+		case "complete":
+			return <CheckCircle sx={{ color, fontSize: 18 }} />;
+		case "failure":
+			return <ErrorIcon sx={{ color, fontSize: 18 }} />;
+		case "working":
+			return <CircularProgress size={18} sx={{ color }} />;
+		default:
+			return <Warning sx={{ color, fontSize: 18 }} />;
+	}
+};
+
+const commonStyles = {
+	fontSize: "12px",
+	lineHeight: "15px",
+	color: "text.disabled",
+	display: "inline-flex",
+	alignItems: "center",
+	gap: 0.5,
+	px: 0.75,
+	py: 0.25,
+	borderRadius: "6px",
+	bgcolor: "transparent",
+	minWidth: 0,
+	maxWidth: "fit-content",
+	overflow: "hidden",
+	textOverflow: "ellipsis",
+	whiteSpace: "nowrap",
+	textDecoration: "none",
+	transition: "all 0.15s ease-in-out",
+	"&:hover": {
+		textDecoration: "none",
+		bgcolor: "action.hover",
+		color: "text.secondary",
+	},
+	"& .MuiSvgIcon-root": {
+		// Consistent icon styling within links
+		fontSize: 11,
+		opacity: 0.7,
+		mt: "-1px", // Slight vertical alignment adjustment
+		flexShrink: 0,
+	},
+};
+
+const formatURI = (uri: string) => {
```

Comment:
There's a lot of util-esque functionality in here I think we should eventually extract (like the short URIs), but don't want that to block right now.